solr/core/src/java/org/apache/solr/cloud/ZkConfigSetService.java

```diff
@@ -276,20 +270,11 @@ public void downloadConfig(String configName, Path dir) throws IOException {
       throw new IllegalArgumentException("\"" + zkPath + "\" not recognized as a configset path");
     }
     try {
-      return zkClient.getAllConfigsetFiles(zkPath);
-    } catch (KeeperException | InterruptedException e) {
-      throw new IOException("Error getting all configset files", SolrZkClient.checkInterrupted(e));
-    }
-  }
-
-  @Override
-  public String getConfigPath(String configName) throws IOException {
-    try {
-      if (zkClient.exists(CONFIGS_ZKNODE + "/" + configName, true)) {
-        return CONFIGS_ZKNODE + "/" + configName;
-      } else {
-        throw new IOException("config does not exist");
+      List<String> filePaths = new LinkedList<>();
+      for (String filePath : zkClient.getAllConfigsetFiles(zkPath)) {
+        filePaths.add(filePath.replaceAll(zkPath + "/", ""));
```

Comment:
I think this will replace any substring of the filePath?  I think we want to specifically chop off the beginning; no?  I'd prefer you assert that each filePath starts with what you think it does, then truncate it off.

Also, I don't see why SolrZkClient needs a getAllConfigsetFiles method.  From what I see, we could call `org.apache.solr.common.cloud.ZkMaintenanceUtils#traverseZkTree` which does *not* prefix it's results with the zkPath argument you give it.

But morover